builder/elfpatch.go

```diff
@@ -0,0 +1,57 @@
+package builder
+
+import (
+	"debug/elf"
+	"fmt"
+	"os"
+)
+
+func getElfSectionData(executable string, sectionName string) ([]byte, elf.FileHeader, error) {
+	elfFile, err := elf.Open(executable)
```

Comment:
You might consider moving the opening of ELF files to some place earlier, so that it is only opened once instead of for each call to `getElfSectionData` and `replaceElfSection`. Optional, the current code looks good and is certainly easy to understand/modify.

Reply:
Yeah - i was undecided about 'going too far' on something that wraps up `debug/elf` with editing capability.  I'll leave it like it is for now and can make the wrapper more complete later if needed.